frontend/src/components/courses/evaluations/Evaluate.tsx

```diff
@@ -0,0 +1,257 @@
+                });
+            });
+    }
+
+    const validateComments = (comments, errors) => {
+        for (let i in comments) {
+            let question: Question = getQuestionById(i);
+            let comment: string = comments[i];
+            if (question.required && (comment === "")) {
+                errors[i] = `${question.label} missing required field`;
+                newToast({
+                    message: `La question intitulée : ${question.label} doit être remplit!`,
+                    level: ToastLevel.Error,
+                });
+            }
+        }
+    }
+
+    const validateRatings = (ratings, errors) => {
+        for (let i in ratings) {
+            let question: Question = getQuestionById(i);
+            let rating: number = ratings[i];
+            if (question.required && (rating === -1)) {
+                errors[i] = `${question.label} missing required field`;
+                newToast({
+                    message: `La question intitulée : ${question.label} doit être remplit!`,
+                    level: ToastLevel.Error,
+                });
+            }
+        }
+    }
+
+    const validate = (values) => {
+        {/* Validate the data */ }
+        const errors = {};
+        validateRatings(values.ratings, errors);
+        validateComments(values.comments, errors);
+        return errors;
+    }
+
+    if (hasVoted) return <Redirect to={`/cours/${course.id}`} />
+    return (
+        <Row>
+            <Formik
+                initialValues={initRating(questions)}
+                onSubmit={submitAnswers}
+                validate={validate}
+            >
+                {(props: FormikProps<Values>) => (
+                    <Form onSubmit={props.handleSubmit}>
+                        {questions.map((question: Question) => {
+                            if (question.archived) return null;
+
+                            let field: JSX.Element = <p>Error</p>;
+                            if (question.category === "R") {
+                                field = <RatingField question={question} id={question.id} name="ratings" label="First Name" {...props} />;
+                            }
+                            else if (question.category === "C") {
+                                field = <CommentField question={question} id={question.id} name="comments" label="First Name" {...props} />;
+                            };
+
+                            let bg = 'light';
+                            if (question.id) if (props.errors[question.id] != undefined) bg = 'danger';
+
+                            return (
+                                <Col md={8} key={question.id}>
+                                    <Card
+                                        key={questions.id}
+                                        className="col-md-4 m-4"
+                                        text={bg === 'light' ? 'dark' : 'white'}
+                                        // @ts-ignore
```

Comment:
Why  ?

Reply:
Replaced with ternary condition and removed the ignore